vault/external_tests/identity/login_mfa_okta_test.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package identity
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openbao/openbao/api"
+	"github.com/openbao/openbao/helper/constants"
+	"github.com/openbao/openbao/builtin/credential/userpass"
+	"github.com/openbao/openbao/helper/testhelpers"
+	logicaltest "github.com/openbao/openbao/helper/testhelpers/logical"
+	vaulthttp "github.com/openbao/openbao/http"
+	"github.com/openbao/openbao/sdk/logical"
+	"github.com/openbao/openbao/vault"
+)
+
+// To run these tests, set the following env variables:
+// VAULT_ACC=1
+// OKTA_ORG=dev-219337
```

Comment:
This is one of the reasons I was originally pushing for open-only tests :-) It is hard for those of us without Okta org accounts to run this integration test suite. It doesn't look like there's a mock Okta-for-testing container officially published, but putting this in an external plugin makes more sense to me (as it can then be updated and fixed independently of the core binary).

But I do also understand that MFA hasn't typically been plugin-ized, so it is probably hard to do. :-)

Reply:
Yeah. This is an ugly test setup.

Creating a dev org/account with okta is free (free as in "no charge"). Maybe there's a way to create an org just for openbao's usage in CI, and put the relevant secrets in GHA? The test in question could also run only on a schedule (once a week maybe) to avoid hitting API usage limits.

The biggest gotcha, though, is having a device with the okta app installed. I don't think there's an android runner in GHA, so I'm not sure how that would work.

Reply:
My favorite hack is that the MacOS runner supports virtualization, so if you can fully script the setup, you could put Android in a VM... :laughing: I don' recommend that idea though.

Reply:
Go figure: there _are_ ways to run an android emulator in GHA.
https://www.atkinsondev.com/post/android-emulator-tests-github-actions/
https://github.com/ReactiveCircus/android-emulator-runner

So, then we'd have to figure out how to install and automate the okta app.